Document that persister must take logical endpoint name into account when storing outbox records (#7137)
Particular/docs.particular.net: nservicebus/outbox/index.md

## Message identity

> The outbox uses the incoming [message identifier](/nservicebus/messaging/message-identity.md) as a unique key for deduplicating messages. If the sender [specifies a custom message id](/nservicebus/messaging/message-identity.md#specifying-message-identity) when sending messages, it is responsible for ensuring that the message identifier value is consistent if the message is sent multiple times.

Reviewer: ```suggestion
The outbox uses the incoming [message identifier](/nservicebus/messaging/message-identity.md) as a unique key for deduplicating messages. When sending messages with a [custom message id](/nservicebus/messaging/message-identity.md#specifying-message-identity), the sender should ensure that the message identifier value is consistent if the message is sent multiple times.
```

> Since message identifier are [only unique with in the scope of a logical endpoint from a processing perspective](/nservicebus/messaging/message-identity.md) its the responsibility of the persister to ensure that message identities are scoped to the [logical endpoint](/nservicebus/endpoints/#logical-endpoints) that is processing the message to allow multiple endpoints to use the same database.

Reviewer: ```suggestion
Since message identifiers are [unique only within the scope of a logical endpoint from a processing perspective](/nservicebus/messaging/message-identity.md) it is the responsibility of the persister to ensure that message identities are scoped to the [logical endpoint](/nservicebus/endpoints/#logical-endpoints) that is processing the message to allow multiple endpoints to use the same database.
```
